exercises/triangle/examples/src/Triangle.purs

```diff
@@ -0,0 +1,41 @@
+module Triangle
+  ( Triangle(Equilateral, Isosceles, Scalene)
+  , triangleKind
+  ) where
+
+import Prelude
+import Data.Set as Set
+import Data.Either (Either(Left, Right))
+import Data.Generic (class Generic, gShow, gEq)
+
+
+data Triangle
+  = Equilateral
+  | Isosceles
+  | Scalene
+
+derive instance genericTriangle :: Generic Triangle
```

Comment:
Yes, there is an explanation in my [24 days of PureScript](https://github.com/paf31/24-days-of-purescript-2016) blog post series (see the Generic deriving post, or Eq and Ord deriving). Also if you only need `eq`, don't use `gEq`, use the `derive` keyword to derive a more efficient `Eq` instance.